### DISCUSSION TITLE
mssqldef: Precision and scale of numeric columns are not output

### DISCUSSION THREAD

**Discussion by odz** [sqldef/sqldef#624] [Feature requests]
## Platform

* OS: Linux
* RDBMS: SQL Server
* Version: 0.16.7

## Crurrent Schema

```sql
CREATE TABLE numbers (
  a numeric,
  b numeric(10),
  c numeric(10, 3)
);
```

## Current output (--export)

```sql
CREATE TABLE dbo.numbers (
    [a] numeric,
    [b] numeric,
    [c] numeric
);
GO
```

## Expected output (--export)

```sql
CREATE TABLE dbo.numbers (
    [a] numeric(5),
    [b] numeric(10),
    [c] numeric(10, 3)
);
GO

```

**Comment by chi-bd** [comment]
@odz
I believe this issue is solved by #568, but ...

> Expected output (--export)
> CREATE TABLE dbo.numbers (
>     [a] numeric(5),

why do you think the default precision of "5" ?
I implemented the default precision of 18, according to the explanation in the following URL.

https://learn.microsoft.com/en-us/sql/t-sql/data-types/decimal-and-numeric-transact-sql?view=sql-server-ver16#p-precision
> decimal [ ( p [ , s ] ) ] and numeric [ ( p [ , s ] ) ]
> p (precision)
> ...
> The default precision is 18.

**Comment by gfx** [comment]
There have been issues about precisions in mssqldef and mysqldef. 

Fixed in https://github.com/sqldef/sqldef/pull/897
